pandas/tests/io/parser/test_parse_dates.py

```diff
@@ -0,0 +1,835 @@
+    parser = all_parsers
+    data = "06-02-2013;13:00;1-000.215"
+    expected = DataFrame([[datetime(2013, 6, 2, 13, 0, 0), 1000.215]],
+                         columns=["Date", 2])
+
+    df = parser.read_csv(StringIO(data), sep=";", thousands="-",
+                         parse_dates={"Date": [0, 1]}, header=None)
+    tm.assert_frame_equal(df, expected)
+
+
+@pytest.mark.parametrize("keep_date_col", [True, False])
+def test_multiple_date_col_custom(all_parsers, keep_date_col):
+    data = """\
+KORD,19990127, 19:00:00, 18:56:00, 0.8100, 2.8100, 7.2000, 0.0000, 280.0000
+KORD,19990127, 20:00:00, 19:56:00, 0.0100, 2.2100, 7.2000, 0.0000, 260.0000
+KORD,19990127, 21:00:00, 20:56:00, -0.5900, 2.2100, 5.7000, 0.0000, 280.0000
+KORD,19990127, 21:00:00, 21:18:00, -0.9900, 2.0100, 3.6000, 0.0000, 270.0000
+KORD,19990127, 22:00:00, 21:56:00, -0.5900, 1.7100, 5.1000, 0.0000, 290.0000
+KORD,19990127, 23:00:00, 22:56:00, -0.5900, 1.7100, 4.6000, 0.0000, 280.0000
+"""
+    parser = all_parsers
+
+    def date_parser(*date_cols):
+        """
+        Test date parser.
+
+        Parameters
+        ----------
+        date_cols : args
+            The list of data columns to parse.
+
+        Returns
+        -------
+        parsed : Series
+        """
+        return parsing.try_parse_dates(parsers._concat_date_cols(date_cols))
+
+    result = parser.read_csv(StringIO(data), header=None,
+                             date_parser=date_parser, prefix="X",
+                             parse_dates={"nominal": [1, 2],
+                                          "actual": [1, 3]},
+                             keep_date_col=keep_date_col)
+    expected = DataFrame([
+        [datetime(1999, 1, 27, 19, 0), datetime(1999, 1, 27, 18, 56),
+         "KORD", "19990127", " 19:00:00", " 18:56:00",
+         0.81, 2.81, 7.2, 0.0, 280.0],
+        [datetime(1999, 1, 27, 20, 0), datetime(1999, 1, 27, 19, 56),
+         "KORD", "19990127", " 20:00:00", " 19:56:00",
+         0.01, 2.21, 7.2, 0.0, 260.0],
+        [datetime(1999, 1, 27, 21, 0), datetime(1999, 1, 27, 20, 56),
+         "KORD", "19990127", " 21:00:00", " 20:56:00",
+         -0.59, 2.21, 5.7, 0.0, 280.0],
+        [datetime(1999, 1, 27, 21, 0), datetime(1999, 1, 27, 21, 18),
+         "KORD", "19990127", " 21:00:00", " 21:18:00",
+         -0.99, 2.01, 3.6, 0.0, 270.0],
+        [datetime(1999, 1, 27, 22, 0), datetime(1999, 1, 27, 21, 56),
+         "KORD", "19990127", " 22:00:00", " 21:56:00",
+         -0.59, 1.71, 5.1, 0.0, 290.0],
+        [datetime(1999, 1, 27, 23, 0), datetime(1999, 1, 27, 22, 56),
+         "KORD", "19990127", " 23:00:00", " 22:56:00",
+         -0.59, 1.71, 4.6, 0.0, 280.0],
+    ], columns=["nominal", "actual", "X0", "X1", "X2",
+                "X3", "X4", "X5", "X6", "X7", "X8"])
+
+    if not keep_date_col:
+        expected = expected.drop(["X1", "X2", "X3"], axis=1)
+    elif parser.engine == "python":
+        expected["X1"] = expected["X1"].astype(np.int64)
```

Comment:
So this is an inconsistency in how we have columns that are aggregated to create one date column.  The C engine does not continue to parse them and infer dtype, whereas the Python does parse and infer dtypes on these columns as if they were regular data columns.

Which engine is more correct?

Reply:
can you open an issue here for followup

Reply:
Opened #23845